.github/workflows/continuous-integration.yml

```diff
@@ -1,11 +1,11 @@
-name: "Test Nickel building on master"
+name: Continuous integration
 on:
-  push:
-    branches:
-      - master
+  - push
+  - pull_request
```

Comment:
Perhaps `push` suffices here?

If you trigger both on `push` and `pull_request` you'll run the workflow twice, for example when you open a pull-request.

I believe this is why results are duplicated in this PR too:
![image](https://user-images.githubusercontent.com/514775/115231213-95877600-a115-11eb-84ea-95c6875e9f35.png)

Reply:
Was wondering the same thing. Thanks for the info!

Reply:
That's useful for external contributions though, because for these only the `pull_request` workflow will show on the PR

Reply:
> That's useful for external contributions though, because for these only the pull_request workflow will show on the PR

Good point. So maybe `push: master` and `pull_request` were an OK approximation.

 @mboes , is it really a problem in practice if topic branches without associated pull requests do not trigger the CI? Anything that will be merged probably has to go through a PR eventually, and opening a Draft PR for ongoing work on a topic branch although doesn't sound like a bad practice. In that case we could go back to `push: master` and `pull_request`, avoiding double checks but still firing on external contributions.

Reply:
@yannham Yeah, that's fine by me. It's a reasonable compromise. We do lose push builds that way, but CI is easy enough to reliably run on one's own machine, and as you say, draft PR's do exist.

Reply:
> That's useful for external contributions though, because for these only the `pull_request` workflow will show on the PR

Is it though? I assume that the push Github action will run on their fork, won't it?

Reply:
It will (unless they disable it I guess), but this won't show in the PR. For example compare https://github.com/NixOS/nix/pull/4511 (made from a branch in the same repo) and https://github.com/NixOS/nix/pull/4060 (made from a branch in a fork)

Reply:
I see. It's a pity :slightly_frowning_face: